src/components/Todos/Header.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+  Dispatch, FormEvent, SetStateAction, useEffect, useRef,
+} from 'react';
+import { createTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  userId: number;
+  todos: Todo[];
+  setTodos: Dispatch<SetStateAction<Todo[]>>;
+  setError: Dispatch<SetStateAction<boolean>>;
+  setErrorMessage: Dispatch<SetStateAction<string>>;
+  setLoader: Dispatch<SetStateAction<boolean>>;
+  title: string;
+  setTitle: Dispatch<SetStateAction<string>>;
+};
+
+export const Header: React.FC<Props> = ({
+  userId,
+  todos,
+  setTodos,
+  setError,
+  setErrorMessage,
+  setLoader,
+  title,
+  setTitle,
+}) => {
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    // focus the element with `ref={newTodoField}`
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, []);
+
+  const handleSumbit = (event: FormEvent) => {
+    event.preventDefault();
+    const normalizedTitle = title.trim();
+
+    if (title.length < 1) {
+      setError(true);
+      setErrorMessage("Title can't be empty");
+
+      return;
+    }
+
+    createTodo(userId, normalizedTitle)
+      .then(addedTodo => {
+        setTodos([...todos, addedTodo]);
```

Comment:
```suggestion
        setTodos(todos => [...todos, addedTodo]);
```

Reply:
`'todos' is already declared in the upper scope`